scripts/zones/The_Garden_of_RuHmet/mobs/Jailer_of_Faith.lua

```diff
@@ -7,27 +7,114 @@ mixins = {require("scripts/mixins/job_special")}
+    mob:setLocalVar("[faith]changeTime", mob:getBattleTime() + math.random(20, 40))
+    mob:setAnimationSub(1)
+end
+
+entity.onMobInitialize = function(mob)
+    mob:addListener("TAKE_DAMAGE", "FAITH_TAKE_DAMAGE", function(mobArg, amount, attacker, attackType, damageType)
+        if attackType == xi.attackType.PHYSICAL then
+            mobArg:setLocalVar("PhysicalDamage", mobArg:getLocalVar("PhysicalDamage") + amount)
+        elseif attackType == xi.attackType.MAGICAL then
+            mobArg:setLocalVar("MagicalDamage", mobArg:getLocalVar("MagicalDamage") + amount)
+        elseif attackType == xi.attackType.RANGED then
+            mobArg:setLocalVar("RangedDamage", mobArg:getLocalVar("RangedDamage") + amount)
+        elseif attackType == xi.attackType.BREATH then
+            mobArg:setLocalVar("BreathDamage", mobArg:getLocalVar("BreathDamage") + amount)
+        else
+            -- ignore Untyped Damage
+        end
+    end)
+end
+
 entity.onMobSpawn = function(mob)
+    mob:setLocalVar("PhysicalDamage", 0)
+    mob:setLocalVar("MagicalDamage", 0)
+    mob:setLocalVar("RangedDamage", 0)
+    mob:setLocalVar("BreathDamage", 0)
+    xi.mix.jobSpecial.config(mob, {
+        specials =
+        {
+            {
+                id = xi.jsa.MANAFONT,
+                endCode = function(mobArg) -- Uses Manafont and casts Quake II -- only casts this during manafont.
+                    mobArg:castSpell(211) -- quake II
+                end,
+            },
+        },
+    })
     -- Change animation to open
-    mob:setAnimationSub(2)
+    mob:AnimationSub(2)
+    mob:setMobMod(xi.mobMod.HP_STANDBACK, 0)
+    mob:setMod(xi.mod.FASTCAST, 30) -- taken from timing of spells from multiple captures
+    mob:setMod(xi.mod.DEF, 450)
 end
 
 entity.onMobFight = function(mob)
-    -- Forms: 0 = Closed  1 = Closed  2 = Open 3 = Closed
-    local randomTime = math.random(45, 180)
-    local changeTime = mob:getLocalVar("changeTime")
-
-    if mob:getBattleTime() - changeTime > randomTime then
-        -- Change close to open.
-        if (mob:getAnimationSub() == 1) then
-            mob:setAnimationSub(2)
-        else -- Change from open to close
-            mob:setAnimationSub(1)
+    mob:addListener("COMBAT_TICK", "FAITH_CTICK", function(mobArg)
+        local sum = mobArg:getLocalVar("PhysicalDamage") + mobArg:getLocalVar("MagicalDamage") + mobArg:getLocalVar("RangedDamage") + mobArg:getLocalVar("BreathDamage")
+        if mobArg:getAnimationSub() == 2 and sum > 1500 then -- Faith will close flower upon taking 1500 damage combined.
+            closeFlower(mobArg)
+        elseif mobArg:getAnimationSub() <= 1 and mobArg:getBattleTime() > mobArg:getLocalVar("[faith]changeTime") then
+            openFlower(mobArg)
+        else
+            -- if no dmg taken - dont trigger a change
         end
-        mob:setLocalVar("changeTime", mob:getBattleTime())
+    end)
+end
+
+entity.onMagicCastingCheck = function(mob, target, spell)
```

Comment:
did you mean `onMobMagicPrepare()` ?

Reply:
When converting to LSB base I actually didn't find any use of that specific function. So I looked around and found that Promathia's original LSB implementation used: onMagicCastingCheck so I went with that.

Reply:
quote possibly a mistake there, or severely undocumented functionality. 

onMobMagicPrepare tho, seems to be used precisely the way you want in many existing places:
https://github.com/LandSandBoat/server/search?p=2&q=onMobMagicPrepare

Reply:
I'll adjust both then.